File: backend/voting-app/src/main/java/com/tu/votingapp/dto/response/survey/SurveyDTO.java
package com.tu.votingapp.dto.response.survey;

import com.tu.votingapp.enums.SurveyStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class SurveyDTO {
    private Long id;
    private String title;
    private String description;
    private Date startDate;
    private Date endDate;
    // Representing the enum value as a String; adjust if needed.
    private SurveyStatus status;
    // List of nested survey questions
    private List<SurveyQuestionDTO> questions;
    // Reference to the ID of the user who created the survey
    private Long createdById;
}

File: frontend/android/app/src/main/java/com/example/android/dto/response/survey/SurveyDTO.kt
package com.example.android.dto.response.survey

import com.example.android.constants.SurveyStatus
import kotlinx.serialization.Serializable

@Serializable
data class SurveyDTO(
    val id: Long? = null,
    val title: String,
    val description: String? = null,
    val startDate: String, // use ISO-8601 format (e.g. "2025-04-19")
    val endDate: String,
    val status: SurveyStatus,
    val questions: List<SurveyQuestionDTO> = emptyList(),
    val createdById: Long? = null
)